test/table.js

```diff
@@ -1378,6 +1378,233 @@ describe('Bigtable/Table', function() {
+        done();
+      });
+    });
+
+    describe('retries', () => {
+      let clock;
+      let setTimeoutSpy;
+      let clearTimeoutSpy;
+      let checkConsistencySpy;
+      let responses;
+
+      beforeEach(() => {
+        clock = sinon.useFakeTimers({
+          toFake: ['setTimeout', 'clearTimeout'],
+        });
+        setTimeoutSpy = sinon.spy(global, 'setTimeout');
+        clearTimeoutSpy = sinon.spy(global, 'clearTimeout');
+        checkConsistencySpy = sinon.spy(table, 'checkConsistency');
+      });
+
+      afterEach(() => {
+        clock.restore();
+      });
+
+      it('should return true if token is consistent', function(done) {
+        responses = [
+          (config, callback) =>
+            callback(null, [{consistency_token: 'sample-token12345'}]),
+          (config, callback) => callback(null, {consistent: true}),
+        ];
+
+        table.bigtable.request = function(config, callback) {
+          responses.shift()(config, callback);
+        };
+
+        table.waitForReplication(function(err, response) {
+          // Checks that a 10 minute timer was set.
+          setTimeoutSpy.calledWith(sinon.match.func, 10 * 60 * 1000);
+
+          // check checkConsistencySpy called for first time
+          assert.strictEqual(checkConsistencySpy.callCount, 1);
+
+          // Checks that clearInterval was called.
+          assert.strictEqual(clearTimeoutSpy.callCount, 1);
+
+          assert.strictEqual(response, true);
+          assert.ifError(err);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should retry checkConsistency', done => {
+        responses = [
+          (config, callback) =>
+            callback(null, [{consistency_token: 'sample-token12345'}]),
+          (config, callback) => callback(null, {consistent: false}),
+          (config, callback) => callback(null, {consistent: true}),
+        ];
+
+        table.bigtable.request = function(config, callback) {
+          responses.shift()(config, callback);
+        };
+
+        table.waitForReplication(function(err, response) {
+          // Checks that a 10 minute timer was set.
+          setTimeoutSpy.calledWith(sinon.match.func, 10 * 60 * 1000);
+
+          // check checkConsistencySpy called for first time
+          checkConsistencySpy.callOnce;
+
+          setTimeoutSpy.calledWith(sinon.match.func, 5000);
+
+          // check checkConsistencySpy called twice after 5seconds
+          clock.tick(5010);
+          assert.strictEqual(checkConsistencySpy.callCount, 2);
+
+          // Checks that clearInterval was called.
+          setTimeoutSpy.called;
+          assert.ifError(err);
+          assert.strictEqual(response, true);
+          done();
+        });
+
+        clock.runAll();
+      });
+
+      it('should return false after 10 min if inconsistency repeats', done => {
+        this.timeout(10 * 61 * 1000); // 10 minutes
+        table.bigtable.request = function(config, callback) {
```

Comment:
Use the responses pattern the other tests are using

Reply:
@kolodny here table.bigtable.request (with congig.method == checkConsistency) will be called unknown n-times, in-order to check, for continuous failure it must return false after 10 mins.

So can not user responses pattern.